minerva/tool_utils.py

```diff
@@ -37,10 +37,19 @@ def parse_tool_call(message: str, tools: dict[str, GenericToolFn]) -> ToolCall:
   Convert arguments to the correct types according to the tool's signature.
   """
 
-  tool_call = (
+  # Support robust argument parsing including quoted strings (single/double),
+  # escaped quotes with backslash, and multiline content inside quotes.
+  quoted_single = cast(Any, pp.QuotedString("'", escChar="\\", multiline=True))
+  quoted_double = cast(Any, pp.QuotedString('"', escChar="\\", multiline=True))
+  simple_token = cast(Any, pp.Word(pp.alphanums + "._-:/"))
+
+  arg_expr = quoted_single | quoted_double | simple_token
+
+  tool_call = cast(
+    Any,
```

Comment:
It would be great if you can add a unit-test to show which examples this code helps solve.

Reply:
@1mTaKoW, yes, it would be great to do add the tests in the same PR we make the change to show why the change is needed.